base/src/main/java/com/smartdevicelink/streaming/video/VideoStreamingParameters.java

```diff
@@ -218,13 +231,10 @@ public void update(VideoStreamingCapability capability, String vehicleMake) {
 
         // This should be the last call as it will return out once a suitable format is found
         final List<VideoStreamingFormat> formats = capability.getSupportedFormats();
-        if (formats != null && formats.size() > 0) {
-            if (this.format != null && formats.contains(this.format)) {
-                return; // given format is supported, so no need to change.
-            }
-            for (VideoStreamingFormat format : formats) {
-                for (VideoStreamingFormat currentlySupportedFormat : currentlySupportedFormats) {
-                    if (currentlySupportedFormat.equals(format)) {
+        if(formats != null && formats.size()>0){
```

Comment:
I feel we should add this check back in, if the format is already set we do not need to iterate over the rest of them
```suggestion
        if(formats != null && formats.size()>0){
            if (this.format != null && formats.contains(this.format)) {
                return; // given format is supported, so no need to change.
            }
```

Reply:
Hello @kostyaBoss for this change you have the same if statement twice now
```
if (formats != null && formats.size() > 0) {
            if (this.format != null && formats.contains(this.format)) {
                return; // given format is supported, so no need to change.
            }
        }

        if(formats != null && formats.size()>0){
            for(VideoStreamingFormat format : formats){
                for(int i = 0; i < currentlySupportedFormats.length; i ++){
                    if(currentlySupportedFormats[i].equals(format) ){
                        this.format = format;
                        return;
                    }
                }
            }
DebugTool.logWarning(TAG, "The VideoStreamingFormat has not been updated because none of the provided formats are supported.");

            //TODO In the future we should set format to null, but might be a breaking change
            // For now, format will remain whatever was set prior to this update
}
``` 

and can be simplified 

```
if (formats != null && formats.size() > 0) {
            if (this.format != null && formats.contains(this.format)) {
                return; // given format is supported, so no need to change.
            }
            for (VideoStreamingFormat format : formats) {
                for (VideoStreamingFormat currentlySupportedFormat : currentlySupportedFormats) {
                    if (currentlySupportedFormat.equals(format)) {
                        this.format = format;
                        return;
                    }
                }
            }
            DebugTool.logWarning(TAG, "The VideoStreamingFormat has not been updated because none of the provided formats are supported.");

            //TODO In the future we should set format to null, but might be a breaking change
            // For now, format will remain whatever was set prior to this update
        }
```